components/datasets/form/DatasetsForm.js

```diff
@@ -130,20 +147,45 @@ class DatasetsForm extends React.Component {
               : ['connectorUrlHint', 'authorization']
           };
 
-          const bodyObj = omit(form, requestOptions.omit);
+          let bodyObj = omit(form, requestOptions.omit);
 
           bodyObj.subscribable = {};
-          form.subscribable.map((s) => {
-            bodyObj.subscribable[s.type] = Object.assign(
+          form.subscribable.forEach((_subscription) => {
+            bodyObj.subscribable[_subscription.type] = Object.assign(
               {},
               {
-                dataQuery: s.dataQuery,
-                subscriptionQuery: s.subscriptionQuery
+                dataQuery: _subscription.dataQuery,
+                subscriptionQuery: _subscription.subscriptionQuery
               }
             );
           });
 
-          // Save the data
+          // every updated dataset will contain set the layer order in its config
+          // if it doesn't previously
+          if (
+            dataset &&
+            layers.length &&
+            (!bodyObj.applicationConfig ||
+            !bodyObj.applicationConfig[process.env.APPLICATIONS] ||
+            !bodyObj.applicationConfig[process.env.APPLICATIONS].layerOrder)
+          ) {
+            bodyObj = {
+              ...bodyObj,
+              applicationConfig: {
```

Comment:
@andresgnlez Couldn't we simplify this a bit using this piece of code instead? 

```
applicationConfig: {
    ...form.applicationConfig,
    [process.env.APPLICATIONS]: {
       ...form.applicationConfig && form.applicationConfig[process.env.APPLICATIONS],
       layerOrder: layers.map(_layer => _layer.id)
    }
}
```

Reply:
Sure. Just make sure it doesn't override anything.